libraries/AP_OSD/AP_OSD_Screen.cpp

```diff
@@ -1380,6 +1397,69 @@ void AP_OSD_Screen::draw_throttle(uint8_t x, uint8_t y)
     backend->write(x, y, false, "%3d%c", gcs().get_hud_throttle(), SYM_PCNT);
 }
 
+void AP_OSD_Screen::draw_sidebars(uint8_t x, uint8_t y)
+{
+    const int8_t total_sectors = 18;
+    static const uint8_t sidebar_sectors[total_sectors] = {
+        SYM_SIDEBAR_A,
+        SYM_SIDEBAR_B,
+        SYM_SIDEBAR_C,
+        SYM_SIDEBAR_D,
+        SYM_SIDEBAR_E,
+        SYM_SIDEBAR_F,
+        SYM_SIDEBAR_G,
+        SYM_SIDEBAR_E,
+        SYM_SIDEBAR_F,
+        SYM_SIDEBAR_G,
+        SYM_SIDEBAR_E,
+        SYM_SIDEBAR_F,
+        SYM_SIDEBAR_G,
+        SYM_SIDEBAR_E,
+        SYM_SIDEBAR_F,
+        SYM_SIDEBAR_H,
+        SYM_SIDEBAR_I,
+        SYM_SIDEBAR_J,
+    };
+
+    // Get altitude and airspeed, scaled to appropriate units
+    float aspd = 0.0f;
+    float alt = 0.0f;
+    AP_AHRS &ahrs = AP::ahrs();
+    WITH_SEMAPHORE(ahrs.get_semaphore());
+    bool have_speed_estimate = ahrs.airspeed_estimate(aspd);
+    if (!have_speed_estimate) { aspd = 0.0f; }
+    ahrs.get_relative_position_D_home(alt);
+    float scaled_aspd = u_scale(SPEED, aspd);
+    float scaled_alt = u_scale(ALTITUDE, -alt);
+    static const int aspd_interval = 10; //units between large tick marks
+    int alt_interval = (osd->units == AP_OSD::UNITS_AVIATION || osd->units == AP_OSD::UNITS_IMPERIAL) ? 20 : 10;
+
+    // render airspeed ladder
+    int aspd_symbol_index = fmodf(scaled_aspd, aspd_interval) / aspd_interval * total_sectors;
+    for (int i = 0; i < 7; i++){
```

Comment:
P.S. notice there's a bit of repetition in the two `for` loops, but probably it's fine as is...